### ISSUE TITLE
Provide an option to disable displaying default values on flags

### ISSUE THREAD

**Issue by ruchika-satalkar** [spf13/cobra#1638]
Some cases require setting a non-zero default values on flag when 0 is a meaningful value to pass to the backend. 
When default value is set to MAXINT or MININT, it should not be displayed in the flag description. 
But I find no way to disable displaying default values on flags

The help message always shows (default: ...) for any optional or required flag.

Is there any way to remove that part of the help message?
Can someone help fix this?
Thanks!

**Comment by jpmcb** [comment]
I don't think this is configurable at cobra's level. This is behavior we inherit from `pflag.FlagUsages` which calls the `FlagUsagesWrapped` function:

https://github.com/spf13/pflag/blob/d5e0c0615acee7028e1e2740a11102313be88de1/flag.go#L731-L737

To make this configurable, we'd need to raise an issue in that repo.

In the mean time, you can simply enough _not_ declare a default value in your command initilization and set the flag latter.

```go
package main

import (
        "math"
        "fmt"

        "github.com/spf13/cobra"
)

var myInt = 0

var MyCmd = &cobra.Command{
        Use: "test",
        Run: myCommand,
}

func init() {
        MyCmd.PersistentFlags().IntVar(&myInt, "number", 0, "A provided int")
}

func myCommand(cmd *cobra.Command, args []string) {
        if myInt == 0 {
                myInt = math.MaxInt
        }

        fmt.Printf("myInt: %v\n", myInt)

}

func main() {
        MyCmd.Execute()
}
```

Running this doesn't produce the max int as any kind of default in the help message but does set it as the default if the user has not provided input:

```
❯ ./example.com -h
Usage:
  test [flags]

Flags:
  -h, --help         help for test
      --number int   A provided int

❯ ./example.com
myInt: 9223372036854775807

❯ ./example.com --number 2
myInt: 2

```

**Comment by rsatalkar** [comment]
Makes sense,

But then in my use-case 0 is still a valid configurable value
so when the user explicitly sets --number 0, this work-around will over-write it to math.MaxInt
